src/psyclone/domain/lfric/kernel/lfric_kernel_metadata.py

```diff
@@ -0,0 +1,506 @@
+        metadata from a kernel in language-level PSyIR.
+
+        :param symbol: the symbol in which the metadata is stored \
+            in language-level PSyIR.
+        :type symbol: :py:class:`psyclone.psyir.symbols.DataTypeSymbol`
+
+        :returns: an instance of LFRicKernelMetadata.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.psyir.\
+            LFRicKernelMetadata`
+
+        :raises TypeError: if the symbol argument is not the expected \
+            type.
+        :raises InternalError: if the datatype of the provided symbol \
+            is not the expected type.
+
+        '''
+        if not isinstance(symbol, DataTypeSymbol):
+            raise TypeError(
+                f"Expected a DataTypeSymbol but found a "
+                f"{type(symbol).__name__}.")
+
+        datatype = symbol.datatype
+
+        if not isinstance(datatype, UnknownFortranType):
+            raise InternalError(
+                f"Expected kernel metadata to be stored in the PSyIR as "
+                f"an UnknownFortranType, but found "
+                f"{type(datatype).__name__}.")
+
+        # In an UnknownFortranType, the declaration is stored as a
+        # string, so use create_from_fortran_string()
+        return LFRicKernelMetadata.create_from_fortran_string(
+            datatype.declaration)
+
+    @staticmethod
+    def create_from_fortran_string(fortran_string):
+        '''Create a new instance of LFRicKernelMetadata populated with
+        metadata stored in a fortran string.
+
+        :param str fortran_string: the metadata stored as Fortran.
+
+        :returns: an instance of LFRicKernelMetadata.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.psyir.\
+            LFRicKernelMetadata`
+
+        :raises ValueError: if the string does not contain a fortran \
+            derived type.
+        :raises ValueError: if kernel metadata is not a Fortran \
+            derived type.
+        :raises ParseError: if meta args is not a list.
+        :raises ParseError: if the metadata has an unexpected format.
+
+        '''
+        kernel_metadata = LFRicKernelMetadata()
+
+        # Ensure the Fortran2003 parser is initialised.
+        _ = ParserFactory().create(std="f2003")
+        reader = FortranStringReader(fortran_string)
+        try:
+            spec_part = Fortran2003.Derived_Type_Def(reader)
+        except Fortran2003.NoMatchError:
+            # pylint: disable=raise-missing-from
+            raise ValueError(
+                f"Expected kernel metadata to be a Fortran derived type, but "
+                f"found '{fortran_string}'.")
+
+        kernel_metadata.name = spec_part.children[0].children[1].tostr()
+
+        # the value of operates on (CELL_COLUMN, ...)
+        value = LFRicKernelMetadata._get_property(
+            spec_part, "operates_on").string
+        kernel_metadata.operates_on = value
+
+        # the value of index offset (NE, ...)
```

Comment:
This comment needs correcting for LFRic.

Reply:
Oops.